runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/endpoints/Endpoint.kt

```diff
@@ -27,9 +23,52 @@ import aws.smithy.kotlin.runtime.http.Url
  * is expected to be mutable and the client cannot modify the endpoint correctly, the operation
  * will likely fail.
  */
-public data class Endpoint(
-    public val uri: Url,
-    public val isHostnameImmutable: Boolean = false,
-) {
+public class Endpoint {
+    /**
+     * The base URL endpoint clients will use to make API calls to e.g. "api.myservice.com".
+     * NOTE: Only `scheme`, `port`, `host` `path`, and `parameters` are valid. Other URL elements are ignored.
+     */
+    public val uri: Url
+
+    /**
+     * Flag indicating whether the hostname can be modified by the SDK client.
+     */
+    public val isHostnameImmutable: Boolean
+
+    /**
+     * A map of additional HTTP headers to be set when making calls against this endpoint.
+     */
+    public val headers: Map<String, List<String>>
```

Comment:
question: Should we re-use our `Headers` type here or is it worthwhile to have this be a `Map`?

Reply:
The typing here was dictated by the SEP, but it's probably acceptable (and preferred) to deviate from strictness in the interest of being more idiomatic, assuming that functionally they're the same (which they are).

Reply:
I think I'm ok either way TBH, was just bringing up the question. If the SEP dictates a map type there may be good reason to not deviate then. Not sure